design/general_refactorings.md

```diff
@@ -0,0 +1,381 @@
+    ```
+
+    * `raise Exception` should be replaced with `raise CredentialsParseException`
+    * `raise RuntimeError` should be replaced with `raise AuthenticationFailedException`
+
+    
+
+* credentials_from_file.py
+
+  * `def load()`
+
+    * `raise Exception` should be replaced with `raise NotLoggedInException`
+
+      ```python
+      if netrc_obj.hosts == {}:
+          raise Exception("You are already logged out")
+      ```
+
+      ```python
+      if netrc_auth == "":
+          raise Exception("You are already logged out")
+      ```
+
+* init_controller.py
+
+  * `def get_server_certificate()` function
+
+    * `RuntimeError` should be replaced with `MissingParameterException`
+
+      ```python
+      if self.conjurrc_data.appliance_url == '':
+          # pylint: disable=raise-missing-from
+          raise RuntimeError("Error: URL is required")
+      ```
+
+    * `RuntimeError` should be replaced with `InvalidURLException`
+
+      ```python
+      if url.scheme != 'https':
+          raise RuntimeError(f"Error: undefined behavior. Reason: The Conjur URL format provided "
+                 f"'{self.conjurrc_data.appliance_url}' is not supported.")
+      ```
+
+    * `RuntimeError` should be replaced with `ConfirmationException` 
+
+      ```python
+      trust_certificate = input("Trust this certificate? (Default=no): ").strip()
+      if trust_certificate.lower() != 'yes':
+          raise RuntimeError("You decided not to trust the certificate")
+      ```
+
+  * `def get_account_info()`
+
+    * `RuntimeError` should be replaced with `MissingParameterException`
+
+      ```python
+      try:
+          self.init_logic.fetch_account_from_server(self.conjurrc_data)
+      # pylint: disable=broad-except,logging-fstring-interpolation
+      except Exception as error:
+          # pylint: disable=line-too-long,logging-fstring-interpolation
+          logging.warning(f"Unable to fetch the account from the Conjur server. Reason: {error}")
+          # If there was a problem fetching the account from the server, we will request one
+          conjurrc_data.account = input("Enter the Conjur account name (required): ").strip()
+      
+          if conjurrc_data.account is None or conjurrc_data.account == '':
+              # pylint: disable=raise-missing-from
+              raise RuntimeError("Error: account is required")
+      ```
+
+  * `def __ensure_overwrite_file()`
+
+    * `Exception` should be replaced with `ConfirmationException`
+
+      ```python
+      force_overwrite = input(f"File {config_file} exists. " \
+                              f"Overwrite? (Default=yes): ").strip()
+      if force_overwrite != '' and force_overwrite.lower() != 'yes':
+          raise Exception(f"Not overwriting {config_file}")
+      ```
+
+* init_logic.py
+
+  * `def get_certificate()`
+
+    * `Exception` should be replaced with `FailedToRetrieveCertificateException` 
```

Comment:
I would highly recommend that we evaluate what other errors the openssl exception throws. At least evaluate the effort it would require to make this addition
I believe this task is not only about refactoring the names of the exceptions but _improving_ our exception handling in general

Reply:
I added it as a bullet so the evaluation will happen during the task. Or you mean it is better be done currently in the design effort?